src/main/java/pl/umk/mat/git2befit/service/workout/factory/implementation/SplitTrainingPlan.java

```diff
@@ -88,65 +93,138 @@ private Training getTrainingForDay(Map<String, List<ExerciseExecution>> training
+                        "CHEST", "BICEPS", "SIXPACK", "BACK"));
+            }
+            case 2 -> {
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "CHEST", "BICEPS", "SIXPACK", "BACK"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "THIGHS", "TRICEPS", "SHOULDERS0", "CALVES"));
+            }
             case 3 -> {
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "CHEST","BICEPS", "SIXPACK"));
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "BACK", "CALVES"));
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "THIGHS", "TRICEPS", "SHOULDERS"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "CHEST", "BICEPS", "SIXPACK"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "BACK", "CALVES"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "THIGHS", "TRICEPS", "SHOULDERS"));
             }
             case 4 -> {
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "CHEST","BICEPS" ));
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "BACK", "CALVES"));
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "THIGHS", "TRICEPS"));
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "SIXPACK","SHOULDERS"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "CHEST", "BICEPS"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "BACK", "CALVES"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "THIGHS", "TRICEPS"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "SIXPACK", "SHOULDERS"));
             }
-            case 5-> {
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "CHEST","BICEPS" ));
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "BACK"));
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "THIGHS"));
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "SIXPACK","SHOULDERS"));
-                trainingList.add(getTrainingForDay(trainingForBodyPart, "CALVES", "TRICEPS" ));
+            case 5 -> {
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "CHEST", "BICEPS"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "BACK"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "THIGHS"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "SIXPACK", "SHOULDERS"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "CALVES", "TRICEPS"));
```

Comment:
Trochę mi śmierdzi takie podejście. Pomyślałbym nad trochę innym rozwiązaniem, a jeśli się nie uda to chociaż powydzielałbym te list gdzieś. Z chęcią poburzomózguję nad tym